conda-store-server/conda_store_server/action/generate_constructor_installer.py

```diff
@@ -52,11 +60,28 @@ def write_file(filename, s):
     # conda and pip need to be in dependencies for the post_install script
     dependencies = ["conda", "pip"]
     pip_dependencies = []
-    for d in specification.dependencies:
-        if type(d) is schema.CondaSpecificationPip:
-            pip_dependencies.extend(d.pip)
-        else:
-            dependencies.append(d)
+    channels = []
+
+    if is_lockfile:
+        # Adds channels
+        channels = [c.url for c in specification.lockfile.metadata.channels]
+
+        # Adds dependencies
+        for p in specification.lockfile.package:
+            if p.manager == "pip":
+                pip_dependencies.append(f"{p.name}=={p.version}")
+            else:
+                dependencies.append(f"{p.name}=={p.version}")
```

Comment:
What is this block for? Are we adding _all_ the packages listed to the database? These are not necessarily the "input packages" for the environment. It's also true that there are no easy ways to fetch that from the `metadata.sources` info, though.

Regardless, this conda spec should also contain the build string, which can be parsed from the URL:

```suggestion
        # Adds dependencies
        for p in specification.lockfile.package:
            if p.manager == "pip":
                pip_dependencies.append(f"{p.name}=={p.version}")
            else:
                ext = ".tar.bz2" if p.url.endswith(".tar.bz2") else ".conda"
                build_string = p.url.split("/")[-1][:-len(ext)].rsplit("-", 2)[-1]
                dependencies.append(f"{p.name}=={p.version}={build_string)")
```

(Review that code, just wrote it on the spot, no testing).